synapse/storage/databases/main/deviceinbox.py

```diff
@@ -633,19 +633,13 @@ def get_all_new_device_messages_txn(
                 " WHERE ? < stream_id AND stream_id <= ?"
                 " GROUP BY destination"
             )
-            txn.execute(sql, (last_id, upper_pos))
+            txn.execute(sql, (last_id, upto_token))
             updates.extend((row[0], row[1:]) for row in txn)
 
             # Order by ascending stream ordering
             updates.sort()
```

Comment:
This is perhaps not needed anymore, but I am unsure that some upper logic is not relying on it so decided to keep it.